client/my-sites/checkout/checkout-thank-you/header.jsx

```diff
@@ -539,8 +541,19 @@ export class CheckoutThankYouHeader extends PureComponent {
 	}
 
 	render() {
-		const { isDataLoaded, isSimplified, hasFailedPurchases, primaryPurchase } = this.props;
-		const classes = { 'is-placeholder': ! isDataLoaded };
+		const {
+			isDataLoaded,
+			isPendingDomainUpdateForTitan,
+			isSimplified,
+			hasFailedPurchases,
+			primaryPurchase,
+			selectedSite,
+		} = this.props;
+		const classes = {
+			'is-placeholder':
+				! isDataLoaded ||
+				( primaryPurchase && isTitanMail( primaryPurchase ) && isPendingDomainUpdateForTitan ),
+		};
```

Comment:
Do we really need to show a placeholder while site domains are being retrieved? My understanding is that it should happen in the background, as this information is not required to display the `Thank You` page.

Reply:
I suppose we could disable the button while we're waiting instead. But if the primary purchase was for Titan, then the only possible action is the Manage Email button, so I am not sure which UX is better, TBH.

Reply:
In my opinion, the `Thank You` page could trigger a reloading of the list of domains but I'm not convinced it should react to it (said differently, it shouldn't care about the state of this list). Most of our pages are built with the premise that if some data is not ready a placeholder should be displayed. That sounds like a good approach to me as otherwise you move the responsibility of loading the data/reacting to changes to other pages. Maybe we should update the `Email` page to display a temporary state while the data loads?

Reply:
You are 100% correct. This is definitely something we should handle in the Email Management component.

I've updated the PR description and testing comments to reflect that I've switched from hacks in the checkout page to a clear and simple check where it should have been all along.